### PR TITLE
IA-2078: from completeness stats table link back to ou linked submissions

### PR DESCRIPTION
**Pull request by codeKameleon**
Add a link in actions column to access the organization unit linked submissions

Related JIRA tickets : [IA-2078](https://bluesquare.atlassian.net/browse/IA-2078?atlOrigin=eyJpIjoiZjAzYjcxZWMwZThlNDk1YjkzMzU2ODI4Y2M3NTI4NmMiLCJwIjoiaiJ9)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- add a `IconButtonComponent` in actions column to link to the organization unit linked submissions
- add message for icon
- update types

## How to test

- Go to Completeness Stats
- If there is at least one submission linked to the form, you should see the eye icon in the action column
- Make sure the link is working

## Print screen / video

https://github.com/BLSQ/iaso/assets/25134301/a46539f5-2188-454b-86e0-96c0cb622c33

[IA-2078]: https://bluesquare.atlassian.net/browse/IA-2078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx [hunk 1]
```diff
@@ -238,10 +239,18 @@ export const useCompletenessStatsColumns = (
...
                             />
                         )}
+                        {settings.row.original.form_stats && (
+                            <IconButtonComponent
+                                id={`form-link-${settings.row.original.id}`}
+                                url={`/${baseUrls.instances}/accountId/${params.accountId}/page/1/tab/list/search/${orgunitName}`}
+                                icon="remove-red-eye"
+                                tooltipMessage={MESSAGES.viewInstances}
+                            />
+                        )}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx]: Please use the Org unit id, as the name is not unique (see other buttons in the action column)

**Review by quang-le — CHANGES_REQUESTED** [review]

**Review by hakifran — APPROVED** [review]
@codeKameleon , a part the comment made by Son, which seems crutial to me too on the remplacement of the orgunitName by the orgunit_id, the rest looks good to me

File: hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx (hunk 1 above)
Comment (hakifran) [reply hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx]: Same comment

Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx]: also i would check if the user has `iaso_submissions` permission

**Review by beygorghor — CHANGES_REQUESTED** [review]

File: hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx [hunk 2]
```diff
@@ -204,6 +211,18 @@ export const useCompletenessStatsColumns = (
...
+                const orgunitId = settings.row.original.org_unit.id;
+                let hasFormSubmissions = false;
+                Object.entries(formStats).forEach(
+                    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+                    ([_key, value]: [string, Record<string, any>]) => {
+                        if (value.itself_has_instances > 0) {
+                            hasFormSubmissions = true;
+                        }
+                    },
+                );
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/completenessStats/hooks/useCompletenessStatsColumns.tsx]: using valuesIn from lodash you can make something like that:
`const hasFormSubmissions = valuesIn(formStats).some(stat => stat.itself_has_instances > 0 );`

**Review by beygorghor — APPROVED** [review]

**Review by beygorghor — APPROVED** [review]

**Review by quang-le — APPROVED** [review]

**Review by hakifran — APPROVED** [review]

**Pull request merged**